content/blog/atomic-design-for-spas.md

```diff
@@ -0,0 +1,213 @@
++++
+title = "Atomic Design in Single Page Applications"
+description = "How I use Atomic Design to architect my SPAs in React"
+date = 2018-10-25
++++
+
+This post will outline how I architect visual part of the SPAs I work on. I used
+to put every components in a `components` folder, which obviously doesn't scale and make
+it hard finding a specific component. After that, I grouped at least the top level routing
+components which made things a bit tidier but was still very unsatisfying.
+Looking around for various ways to organise design elements, I found [Atomic Design](http://bradfrost.com/blog/post/atomic-web-design/).
+I am not going to dive deep into explaning what is Atomatic Design as the linked article does a very good job
+so I would recommend taking 2 minutes to read it beforehand.
+
+You can find a basic clone of [HackerNews](https://news.ycombinator.com/) on GitHub: <https://github.com/Keats/hn-react> made using
+the principles described in this post.
+
+Something missing from the repository that I use in my own projects is [StoryBook](https://storybook.js.org/) to easily have
+a visual overview of every component - an invaluable tool.
+
+Lastly, while the title doesn't mention a specific framework, all the samples will use React. The
+ideas presented in this post will work equally well in Vue.js or any other component-based
+framework.
+
+
+## Self-contained components
+That is something Vue.js users can get behind!
+Each component has now its own folder containing everything it needs to be used, except
+for other components as we will see in a bit.
+
+What this means is that every component folder looks very similar to something like:
+
+```bash
+input
+├── index.tsx
+└── _style.scss
+```
+
+You can replace `tsx` by `js` if you are not using [TypeScript](https://www.typescriptlang.org/).
+
+As you can see, the folder also contains a `_style.scss`. There might be some use-cases for CSS in JS but I still haven't seen a compelling one
```

Comment:
Well the compelling arguments are http://cssinjs.org/benefits/?v=v9.8.7

Reply:
Seen that but don't really agree with them, hence the "compelling"

Reply:
maybe instead of saying "Why do CSS in JS?" you should give an argument (or a few) for having a separate css file